roles/openshift_repos/tasks/main.yaml

```diff
@@ -1,13 +1,20 @@
 ---
+- set_fact:
+    deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+    openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+
+- name: Ensure a valid deployment type has been given
+  fail:
+    msg: "Please set openshift_deployment_type to one of:\n{{ known_openshift_deployment_types | join(', ') }}"
+  when: >
+    openshift_deployment_type is not defined
```

Comment:
openshift_deployment_type will always be defined because of set_fact above (or if moving to defaults).